system-addon/test/functional/mochitest/browser_topsites_contextMenu_options.js

```diff
@@ -25,3 +22,34 @@ test_newtab({
     }
   }
 });
+
+test_newtab({
+  before: setDefaultTopSites,
+  // Test verifies that the next top site in queue replaces a dismissed top site.
+  test: async function defaultTopSites_dismiss() {
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".top-site-icon"),
+      "Topsite tippytop icon not found");
+
+    let defaultTopSitesNumber = content.document.querySelector(".top-sites-list").querySelectorAll("[class=\"top-site-outer\"]").length;
+    Assert.equal(defaultTopSitesNumber, 6, "6 top sites are loaded by default");
+
+    let secondTopSite = content.document.querySelector(".top-sites-list li:nth-child(2) a").getAttribute("href");
+
+    let contextMenuItems = content.openContextMenuAndGetOptions(".top-sites-list li:first-child");
+    Assert.equal(contextMenuItems[4].textContent, "Dismiss", "'Dismiss' is the 5th item in the context menu list");
+
+    contextMenuItems[4].querySelector("a").click();
+
+    // Need to wait for dismiss action.
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".top-sites-list li:first-child a").getAttribute("href") === secondTopSite,
+      "First topsite was dismissed");
+
+    defaultTopSitesNumber = content.document.querySelector(".top-sites-list").querySelectorAll("[class=\"top-site-outer\"]").length;
+    Assert.equal(defaultTopSitesNumber, 5, "5 top sites are displayed after one of them is dismissed");
+  },
+  async after() {
+    await SpecialPowers.pushPrefEnv({set: [["browser.newtabpage.blocked", ""]]});
+    await new Promise(resolve => NewTabUtils.undoAll(resolve));
+    await setDefaultTopSites();
```

Comment:
As noted in the previous comment, fixing up `setDefaultTopSites` to use the scoped one might avoid needing to do this here. If we do need it here for `--verify`, we should figure out why it's needed to figure out a better way to not need it.